### DISCUSSION TITLE
New body property

### DISCUSSION THREAD

**Discussion by Transcan** [EDCD/EDDI#2283] [Needs thought]
Hello there,

I use EDDI a lot for exploration purposes. I already tweaked the personality to my own needs but there is something that I miss in EDDI.

Some context first.
I tweaked "_body scanned_" script in order to inform me when a scanned body is worth for mapping, and also tells me the expected value of that mapping (aka, the maximum exploration value of that body).
I used "**body.estimatedvalue**" for such value, because the manual says it takes in count the scan and the mapping... But it seems that EDDI only gave you the actual state of that particular body.
In other words, if you didn't mapped it yet, you will only get the scan value.

So that doesn't work for my script, at least not as I intended: Tell me the mapping value and decide if its worth enough to make the trip to the body and map it.
Yes, I know its always a similar value for a particular body type, so the prize wouldn't be decisive... But I prefer to hear "Mapping this body will give you more than 2 millions", than "this body will give you 300 thousands".

whatsoever... To the point.

I wanted to see in a future release of EDDI a new body property that gave the maximum obtainable exploration value of a body.
In other words, the same function, but instead of using actual "_alreadyMapped_" state, return the value making it always been "_true_". With this change, the resultant value is the maximum value the commander can get from that body, in its actual state (taking in count the first discovery and first mapping states).
It could be named something like body.maxvalue (or body.maxscanvalue).

Meanwhile I "fixed" it myself by cloning the built-in _estimateBodyValue()_ function into a cottle script, calling it whenever a scan event occurs, and storing the value in a new state variable named "_eddi_context_last_scan_max_value_", along with the usual "_eddi_context_last_scan_estimated_value_".
The scripts uses those variables for their magic, instead the one gave by _body.estimatedvalue_ and all is fine and working as I intended...

But I feel its redundant and I think this new property is quite easy to implement.
What do you think?

**Comment by Tkael** [comment]
Should be possible. I've converted this to a ticket (#2284) to consider for a future update. o7
